ipfs-api-prelude/src/api.rs

```diff
@@ -2024,15 +2024,27 @@ pub trait IpfsApi: Backend {
     /// use ipfs_api::{IpfsApi, IpfsClient};
     ///
     /// let client = IpfsClient::default();
-    /// let res = client.pubsub_pub("feed", "Hello World!");
+    /// let res = client.pubsub_pub("feed", std::io::Cursor::new("Hello World!"));
     /// ```
     ///
-    async fn pubsub_pub(
+    async fn pubsub_pub<T, R>(
         &self,
-        topic: &str,
-        payload: &str,
-    ) -> Result<response::PubsubPubResponse, Self::Error> {
-        self.request_empty(request::PubsubPub { topic, payload }, None)
+        topic: T,
+        data: R,
+    ) -> Result<response::PubsubPubResponse, Self::Error>
+    where
+        T: AsRef<[u8]>,
+        R: 'static + Read + Send + Sync,
+    {
+        use multibase::{encode, Base};
+
+        let topic = encode(Base::Base64Url, topic);
```

Comment:
Would it be possible to do this in serde, similar to how you're handling deserialization?

Reply:
Do I add a serde.rs in requests for the serializer impl ?